packages/service-provider-core/src/printable-bson.ts

```diff
@@ -7,26 +9,73 @@ type BSONClassKey = (typeof BSON)[Exclude<
 
 // Turn e.g. 'new Double(...)' into 'Double(...)' but preserve possible leading whitespace
 function removeNewFromInspectResult(str: string): string {
-  return String(str).replace(/^(\s*)(new )/, '$1');
+  return str.replace(/^(\s*)(new )/, '$1');
+}
+
+/** Typed array such as Int8Array have a format like 'Int8Array(3) [1, 2, 3]'
+ *  and we want to remove the prefix and keep just the array contents. */
+function removeTypedArrayPrefixFromInspectResult(str: string): string {
+  return str.replace(/^\s*\S+\s*\(\d+\)\s*/, '');
 }
 
 // Create a Node.js-util-inspect() style custom inspect function that
 // strips 'new ' from inspect results but otherwise uses the Node.js
 // driver's bson library's inspect functions.
-function makeClasslessInspect<K extends BSONClassKey>(className: K) {
+function makeClasslessInspect<K extends BSONClassKey>(
+  className: K
+): CustomInspectFunction {
   const originalInspect = BSON[className].prototype.inspect;
   return function (
-    this: (typeof BSON)[typeof className]['prototype'],
-    ...args: any
+    this: typeof originalInspect,
+    ...args: Parameters<typeof originalInspect>
   ) {
-    return removeNewFromInspectResult(originalInspect.apply(this, args as any));
-  };
+    return removeNewFromInspectResult(originalInspect.apply(this, args));
+  } as CustomInspectFunction;
 }
 
 const binaryInspect = makeClasslessInspect('Binary');
+
+const binaryVectorInspect = function (
+  this: typeof BSON.Binary.prototype,
+  depth: number,
+  options: InspectOptionsStylized
+): string {
+  switch (this.buffer[0]) {
+    case BSON.Binary.VECTOR_TYPE.Int8:
+      return `Binary.fromInt8Array(new Int8Array(${removeTypedArrayPrefixFromInspectResult(
+        utilInspect(this.toInt8Array(), {
+          depth,
+          ...options,
+          // These arrays can be very large, so would prefer to use the default options instead.
+          maxArrayLength: utilInspect.defaultOptions.maxArrayLength,
+        })
+      )})`;
+    case BSON.Binary.VECTOR_TYPE.Float32:
+      return `Binary.fromFloat32Array(new Float32Array(${removeTypedArrayPrefixFromInspectResult(
+        utilInspect(this.toFloat32Array(), {
+          depth,
+          ...options,
+          // These arrays can be very large, so would prefer to use the default options instead.
+          maxArrayLength: utilInspect.defaultOptions.maxArrayLength,
```

Comment:
not sure if we want this, could remove. it just seems possible that these arrays can be very large I'd imagine so the `{maxArrayLength: Infinity}` that we're getting from https://github.com/mongodb-js/mongosh/blob/d93774d30aa5d64d5cdf52e58915a0fc1e2a031f/packages/cli-repl/src/format-output.ts#L68 can be problematic